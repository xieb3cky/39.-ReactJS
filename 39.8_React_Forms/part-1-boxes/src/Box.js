import React from "react";

function Box({ id, removeBox, width = 5, height = 5, backgroundColor = "pink" }) {
    const remove = () => {
        removeBox(id);
    }
    return (
        <div>
            <div
                style={{
                    height: `${height}em`,
                    width: `${width}em`,
                    backgroundColor: backgroundColor
                }}>
            </div>
            <button onClick={remove}>Remove</button>
        </div >
    );
}



export default Box;