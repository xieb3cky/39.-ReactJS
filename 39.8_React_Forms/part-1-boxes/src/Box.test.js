import { render } from "@testing-library/react";
import React from "react";
import Box from "./Box";

// Smoke Test
it("renders without crashing", function () {
    render(<Box />);
});

//Snapshot Test
it("matches snapshot", function () {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});
