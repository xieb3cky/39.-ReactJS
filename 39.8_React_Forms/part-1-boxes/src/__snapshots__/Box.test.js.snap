// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="height: 5em; width: 5em; background-color: pink;"
    />
    <button>
      Remove
    </button>
  </div>
</DocumentFragment>
`;
